src/runner/mod.rs

```diff
@@ -70,7 +70,9 @@ pub fn run_ex<DB: WriteResults + Sync>(
     // proceed slightly faster than it would otherwise.
     for tc in workspace.installed_toolchains()? {
         // But don't uninstall it if we're going to reinstall in a couple lines.
-        if !ex.toolchains.iter().any(|t| tc == t.source) {
+        // And don't uninstall stable, since that is mainly used for
+        // installing tools (this really should check against `Toolchain::MAIN`).
+        if !tc.to_string().starts_with("stable") && !ex.toolchains.iter().any(|t| tc == t.source) {
```

Comment:
I think we can do `tc.as_dist().map_or(false, |t| t.name() == "stable")` here, but maybe that's not any better.

Reply:
Yeah this is slightly better. Something like `Toolchain::is_needed_by_rustwide()` would be good to add upstream.

Reply:
Sure!  Posted https://github.com/rust-lang/rustwide/pull/64 to add a method.